lang/src/org/partiql/lang/eval/PartiQLResult.kt

```diff
@@ -0,0 +1,19 @@
+package org.partiql.lang.eval
+
+/**
+ * Result of an evaluated PartiQLStatement
+ */
+sealed class PartiQLResult {
```

Comment:
I'm not sure about the error handling patterns here. It seems weird to use error handling patterns from other languages, but maybe there's value to it? Like, in Go you could return `result, err` which could be replicated in Kotlin with Pair like
```kotlin
fun foo(...): Pair<T, Error?> { ... }

val res, err = foo(...)
if (err != null) {
  // handle error
}
```

But I'd prefer to use traditional Java exception handling rather than introduce patterns that were introduced for the semantics of other languages. I've seen this spiral out of control with vavr pattern matching from Scala replicated in Java

Reply:
My original comment was meant to highlight that the term "Result" is used for both `PartiQLPlanner.Result` and `PartiqlResult` but one has an error variant while the other does not:

```Kotlin
// PartiQLPlanner.Result
sealed class Result {

    data class Success(
        val plan: PartiqlPhysical.Plan,
        val warnings: List<Problem>
    ) : Result()

    data class Error(val problems: List<Problem>) : Result() {
        override fun toString(): String = problems.joinToString()
    }
}
```
and 
```Kotlin
// PartiQLResult
sealed class PartiQLResult {

    class Value(val value: ExprValue) : PartiQLResult()

    class Insert(
        val target: String,
        val rows: Iterable<ExprValue>
    ) : PartiQLResult()

    class Delete(
        val target: String,
        val rows: Iterable<ExprValue>
    ) : PartiQLResult()
}
```
This API inconsistency could be annoying in the client code that uses both results. My suggestion would to be consistent when using the "Result" term to either
1. make classes with "Result" include an error variant
2. change the name of one of the "Result" classes to use a different term

Reply:
I think my point here is that Rust error handling patterns were introduced to the Kotlin code base. Not necessarily a bad thing, but is atypical. Im hesitant to introduce other language's patterns into a single place in the code base. PartiQLResult does not need an error variant, in the JVM that's an exception. I kept the Planner Result Success/Error to be consistent with the initial experimental implementation, but my opinion is that it should be removed.

Reply:
PartiQLResult isn't a great name, but I don't have anything better right now. We need a class to represent the result of an evaluated statement

Reply:
I guess the idea at the time for the planner using Result Success/Error was so that it could report multiple errors and/or warnings rather than throwing on the first encountered error. So long as we want the planner to keep the multiple error + warning behavior, I think the naming and usage of Result Success/Error does make sense.

Regarding `PartiQLResult`'s naming, I don't have a great name in mind for what's being described. Perhaps `PartiQLOutput` or `PartiQLEvalOutput`?

Reply:
Reporting multiple _warnings_ seems good, but not multiple errors. I'm of the opinion that you can't continue to do work if a error has already been encountered (because the query is a single expression tree!). You can see in the existing planner that non-sense has to be returned because it attempts to keep working on the tree in an erroneous state. Warning/error collection makes sense for things like parsing other programming languages (antlr's error listeners) because you effectively restart at different statements — however a PartiQL/SQL query is one expression tree, so I would argue that any fatal error invalidates planning. That being said, there are non-fatal errors where some fallaback behavior is chosen, but what's a non-fatal error but a warning?